File: src/main/java/toby/jpa/persistence/IExcusePersistence.java
package toby.jpa.persistence;

import toby.jpa.dto.ExcuseDto;

import java.util.List;

public interface IExcusePersistence {

    List<ExcuseDto> listAllGuildExcuses(Long guildId);
    List<ExcuseDto> listApprovedGuildExcuses(Long guildId);
    List<ExcuseDto> listPendingGuildExcuses(Long guildId);
    ExcuseDto createNewExcuse(ExcuseDto excuseDto);
    ExcuseDto getExcuseById(Integer id);
    ExcuseDto updateExcuse(ExcuseDto excuseDto);
    void deleteAllExcusesForGuild(Long excuseDto);
    void deleteExcuseById(Integer discordId);
}

File: src/main/kotlin/toby/jpa/persistence/IExcusePersistence.kt
package toby.jpa.persistence

import toby.jpa.dto.ExcuseDto

interface IExcusePersistence {
    fun listAllGuildExcuses(guildId: Long?): List<ExcuseDto?>
    fun listApprovedGuildExcuses(guildId: Long?): List<ExcuseDto?>
    fun listPendingGuildExcuses(guildId: Long?): List<ExcuseDto?>
    fun createNewExcuse(excuseDto: ExcuseDto?): ExcuseDto?
    fun getExcuseById(id: Int?): ExcuseDto
    fun updateExcuse(excuseDto: ExcuseDto): ExcuseDto
    fun deleteAllExcusesForGuild(guildId: Long?)
    fun deleteExcuseById(id: Int?)
}
